pkg/chunk/aws/s3_storage_client.go

```diff
@@ -100,7 +108,10 @@ func (cfg *S3Config) RegisterFlagsWithPrefix(prefix string, f *flag.FlagSet) {
 	f.StringVar(&cfg.AccessKeyID, prefix+"s3.access-key-id", "", "AWS Access Key ID")
 	f.StringVar(&cfg.SecretAccessKey, prefix+"s3.secret-access-key", "", "AWS Secret Access Key")
 	f.BoolVar(&cfg.Insecure, prefix+"s3.insecure", false, "Disable https on s3 connection.")
-	f.BoolVar(&cfg.SSEEncryption, prefix+"s3.sse-encryption", false, "Enable AES256 AWS Server Side Encryption")
+	f.BoolVar(&cfg.SSEEncryption, prefix+"s3.sse-encryption", false, "Enable AWS Server Side Encryption [Deprecated: Use sse-config instead. if s3.sse-encryption is enabled, it assumes sse-config SSE-S3 type]")
+	f.StringVar(&cfg.SSEConfig.Type, prefix+"s3.sse-config.type", "", "Enable AWS Server Side Encryption. Only SSE-S3 and SSE-KMS are supported")
```

Comment:
We're used to define `RegisterFlagsWithPrefix(prefix string, f *flag.FlagSet)` on the receiver config. I would suggest you to do the same on `SSEConfig`, passing `prefix+"s3.sse."` as prefix to `SSEConfig.RegisterFlagsWithPrefix()`.